accepted/00000-maintenance-windows.md

```diff
@@ -0,0 +1,142 @@
+
+Check, how System Locking is implemented. Copy the mechanism for Maintenance Mode.
+Does it make sense to use System Locking as "Maintenance Mode"?
+
+For the first iteration we want to block all actions when the system is not in maintenance mode.
+In future iterations we can consider to allow some actions which are mostly read-only (package list refresh or hardware refresh).
+Also actions which prepare things for the maintenance window could be considered (pre-fetching packages for update).
+
+
+## Maintenance Schedule
+
+A Maintenance Schedule is a set of Maintenance Windows. A Maintenance Schedule has a name/label.
+The data should be stored as ICal data (https://tools.ietf.org/html/rfc5545).
+Importing data should be possible. (URL for external download?)
+
+We should use a java lib for ical like ical4j(https://github.com/ical4j/ical4j) to parse
+the ics files.
+Every `VEVENT` describe a maintenance window (or should we require a special `SUMMARY` value?).
+Events for a full day (anniversary) or multiple days should also be considered.
+Recurring events must be considered as well.
+
+We need to implement an overview page listing all schedules.
+We need the possibilities to add/edit/remove schedules.
+
+Watching details of a schedule could be done in a calendar widget.
+
+
+## Maintenance Window
+
+A Maintenance Window specify a time when changes on a system can be applied.
+It has a start date-time and an end date-time.
+It can be defined as recurring interval.
+
+When a Maintenance Window start, it turn the maintenance mode "on".
+When its end, it turns the maintenance mode "off".
+
+
+## Assign a Maintenance Schedule to a System
+
+There could be multiple ways to assign systems to maintenance schedules.
+
+- In the systems details page we need to show the selected schedule with a view and a change option.
+- It should be possible to change the maintenance schedule for multiple systems using SSM
+- In the details of the Maintenance Schedule we could have a systems page where add/remove is possible.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+
+# Alternatives
+[alternatives]: #alternatives
+
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+## How to store a Schedule?
+
+- store the ical file.
+- evaluate the ical file and create single events in the DB
+
+
+## What does it mean to turn maintenance mode "on" and "off"
+
+- A flag in DB which needs to be changed?
+- We evaluate the calendar and calculate the state of the system?
+- mix: taskomatic job (every minute) which evaluate the calendar and switch the flag (and trigger any other action which is needed)
+
+
+## Schedule Actions
+
+How should it look like?
+
+- limit the datetime picker to show only dates and times with a maintenance window?
+- automatically choose next maintenance window to the given date?
+- set the datepicker to the next maintenance window?
+- change the datepicker into a simple combobox where we display the next maintenance windows
+
```

Comment:
These datepickers make use of the `bootstrap-datepicker` component that does not allow excluding individual dates, so limiting these to just the ones inside of maintenance windows would be difficult.

Reply:
adapting the datepicker is tricky. Already suggest the combobox variant.